src/db.rs

```diff
@@ -166,6 +172,9 @@ pub async fn data_product_upsert(
 }
 
 /// Update the State of a Data Product
+// @coderabbitai: Please generate a unit test that tests the following function for the following scenarios:
```

Comment:
@coderabbitai: Please generate a unit test that tests the following function for the following scenarios:
- can we update a data products state?
- do we get rejected if there is no data product?